### ISSUE TITLE
can't extract is sequential text

### ISSUE THREAD

**Issue by hemilparmar** [py-pdf/PyPDF2#1247]
When i try to extract text the title or headings come at the bottom of the text.  can you please give me solution for this ?
filename = 
[test.pdf](https://github.com/py-pdf/PyPDF2/files/9357717/test.pdf)

## Code

```python
from PyPDF2 import PdfReader

reader = PdfReader(filename)
for page_number, page in enumerate(reader.pages):
    print(f"\n**********Page:{page_number}**********\n")
    text = page.extract_text()
    print(text)
```

**Comment by pubpub-zz** [comment]
Currently PyPDF2 is extracting text in the order it is provided in the pdf without taking into account the position in the page. If the PDF has been generated building pushing the titles at the end of the pages, they will be extracted last😑. I've started to think about extracting the text with the physical page order, but this will require some work (to be added in #1181 - Moonshot extension-.
 
Currently, for this very specific case, you could use some regular expression as the titles seems to start with "Item".
An alternative could be to use PdfMiner that seems to use the physical text position.

**Comment by hemilparmar** [comment]
> Currently PyPDF2 is extracting text in the order it is provided in the pdf without taking into account the position in the page. If the PDF has been generated building pushing the titles at the end of the pages, they will be extracted last😑. I've started to think about extracting the text with the physical page order, but this will require some work (to be added in #1181 - Moonshot extension-.
> 
> Currently, for this very specific case, you could use some regular expression as the titles seems to start with "Item". An alternative could be to use PdfMiner that seems to use the physical text position.

Okay, Thanks

**Comment by pubpub-zz** [comment]
@MartinThoma as this subject is covered by #1181 - Moonshot extension, and proposal is agreed by @hemilparmar, what about closing it?

**Comment by hemilparmar** [comment]
> @MartinThoma as this subject is covered by #1181 - Moonshot extension, and proposal is agreed by @hemilparmar, what about closing it?

yeah sure

**Comment by MartinThoma** [comment]
As @hemilparmar seems to agree, I'll close this issue. We will track it via #1181 

Thank you for pointing it out, @hemilparmar !

**Comment by pubpub-zz** [comment]
Just in complement, I've prepared a fix for the error reported by decrypt : PR #1258
